File: backend/src/main/java/Presentacion/DTOS/Usuarios/Register/RegistroBaseDTO.java
package Presentacion.DTOS.Usuarios.Register;

import Dominio.Entity.Roles.Roles;
import lombok.Data;

@Data
public class RegistroBaseDTO {
    private String email;
    private String username;
    private String password;
    private String nombre;
    private int edad;
    private Roles rol;
}

File: frontend/lib/models/DTOS/Registro/registroBaseDTO.dart
// lib/models/DTOS/Registro/registroBaseDTO.dart

import '../../role.dart';

class RegistroBaseDTO {
  String email;
  String username;
  String nombre;
  String apellido;
  int edad;
  String password;
  Role? rol;

  RegistroBaseDTO({
    required this.email,
    required this.username,
    required this.nombre,
    required this.apellido,
    required this.edad,
    required this.password,
    this.rol,
  });

  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'username': username,
      'nombre': nombre,
      'apellido': apellido,
      'edad': edad,
      'password': password,
      'rol': rol?.value ?? 'USUARIO',
    };
  }
}